src/tools/tidy/src/target_specific_tests.rs

```diff
@@ -10,6 +10,25 @@ use crate::walk::filter_not_rust;
 const LLVM_COMPONENTS_HEADER: &str = "needs-llvm-components:";
 const COMPILE_FLAGS_HEADER: &str = "compile-flags:";
 
+const KNOWN_LLVM_COMPONENTS: &[&str] = &[
+    "aarch64",
+    "arm",
+    "avr",
+    "bpf",
+    "hexagon",
+    "loongarch",
+    "m68k",
+    "mips",
+    "msp430",
+    "nvptx",
+    "powerpc",
+    "riscv",
+    "sparc",
+    "systemz",
+    "webassembly",
+    "x86",
+];
```

Comment:
I considered invoking `llvm-config --components` to get this list, but I think that would be a bit awkward to do within tidy.

I think hardcoding it is fine. It should be _very_ rare that this needs to be changed--only when adding support for a brand new architecture.

Reply:
It would mean tidy fails if it can't run llvm-config, and I'm not sure we actually haul that binary around...

Reply:
Would be good to give people some guidance on how to figure out the correct current list.

Reply:
FWIW this works for me
```
build/x86_64-unknown-linux-gnu/ci-llvm/bin/llvm-config --components
```
but it lists a lot more than this PR.

Reply:
many of them are actually the same thing, like this:
- hexagon
- hexagonasmparser
- hexagoncodegen
- hexagondesc
- hexagondisassembler
- hexagoninfo

for our purposes we only really care about having "hexagon"